src/services/sync/strategies/consumerActions/BaseConsumerAction.js

```diff
@@ -0,0 +1,80 @@
+const { NotImplemented, SyncError, BadRequest } = require('../../../../errors');
+const { batchFilterKeys } = require('../../utils');
+
+const securedKeys = ['type', 'getType', 'matchType', 'filterData', 'exec', 'action', 'getSecuredKeys'];
+
+class BaseConsumerAction {
+	constructor(type, options) {
+		if (!type) {
+			throw new NotImplemented('The type must set for consumer actions.');
+		}
+		this.filterActive = true;
+		this.type = type;
+		this.allowedLogKeys = options.allowedLogKeys || null;
+	}
+
+	/**
+	 * @public
+	 * @returns {Array[string]}
+	 */
+	getSecuredKeys() {
+		return securedKeys;
+	}
+
+	/**
+	 * @public
+	 * @returns {string}
+	 */
+	getType() {
+		return this.type;
+	}
+
+	/**
+	 * @private
+	 * @param {string} type
+	 * @returns {boolean}
+	 */
+	matchType(type) {
+		return this.getType() === type;
+	}
+
+	/**
+	 * @private
+	 * @param {object} data
+	 * @returns {Promise}
+	 */
+	async action(data = {}) {
+		throw new NotImplemented('Must be implemented');
+	}
+
+	/**
+	 * @private
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

Reply:
" The proposal is currently a TC39 stage 3: candidate and is likely to be added to ES2019 (ES10). Private fields are currently supported in Node.js 12, Chrome 74, and Babel."

Sound like we can use it. If it go out as candidate we can add it again over the key getter setter if we like.